keps/sig-network/0013-20180612-ipv4-ipv6-dual-stack.md

```diff
@@ -0,0 +1,362 @@
+- --cluster-cidr  string       [Existing argument. Retained for backwords compatibility. Can be an IPv4 or IPv6 CIDR]
+- --cluster-cidrs stringSlice  [New argument. Comma-separated list of CIDRs]
+If both arguments are specified, the CIDRs from both will be combined.
+
+### IPVS Support and Operation
+
+\<TBD\>
+
+### Support of Health/Liveness/Readiness Probes
+
+Currently, health, liveness, and readiness probes are defined without any concern for IP addresses or families. For the first release of dual-stack support, no configuration "knobs" will be added for probe definitions. A probe for a dual-stack pod will be deemed successful if either an IPv4 or IPv6 response is received. (QUESTION: Does the current probe implementation include DNS lookups, or are IP addresses hard coded?)
+
+In the future, we may want to consider adding configuration support to allow a user e.g. to specify whether health probes require:
+- IPv4 connectivity
+- IPv6 connectivity
+- IPv4 or IPv6 connectivity
+- IPv4 and IPv6 connectivity
+
+Or we could offer even more control, e.g. allowing the user to select which particular prefixes should be considered when determining which IPs need to be included for health probes. If we were to offer this capability, we would want to avoid configuring the prefixes directly in the probe definition. Having explicit prefixes in the probe definition would limit its portability from cluster to cluster (there would be a dependence on prefixes that might not exist in other clusters). Instead, the probe definition would need to refer to a prefix indirectly through an abstract tag or label.
+
+### Kube-DNS/CoreDNS Operation
+
+kube-DNS and coreDNS services should be able to support dual-stack services and endpoints with very little change, if any. What is required is that the DNS servers are capable of presenting all IPv4 and IPv6 backend pod addresses as A and AAAA records, respectively, when resolving Kubernetes services.
+
+### Ingress Controller Operation
+
+The [Kubernetes ingress feature](https://kubernetes.io/docs/concepts/services-networking/ingress/) relies on the use of an ingress controller. The two "reference" ingress controllers that are considered here are the [GCE ingress controller](https://github.com/kubernetes/ingress-gce/blob/master/README.md#glbc) and the [NGINX ingress controller](https://github.com/kubernetes/ingress-nginx/blob/master/README.md#nginx-ingress-controller).
+
+#### GCE Ingress Controller: Does Not Support External, Dual-Stack Operation
+It appears that [GCE ingress controller](https://github.com/kubernetes/ingress-gce/blob/master/README.md#glbc) support is limited to IPv4-only access (QUESTION: Is this true???) from outside of the cluster. Therefore, GCE ingress controllers will not be capable of supporting external, dual-stack access to services running in a dual-stack Kubernetes cluster.
```

Comment:
no, we allow V6, but I don't know if we allow both at the same time?  Knowing the data model, I expect we do.